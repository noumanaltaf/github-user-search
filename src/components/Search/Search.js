import React from 'react';
import { Icon, SearchIcon, SearchContainer, SearchInput } from './Search.styled';
import PersonSvg from '../../assets/svg/person.svg';
import SearchSvg from '../../assets/svg/search.svg';

const Search = (props) => {
  const { searchOnClick, ...restInputProps } = props

  return (
    <SearchContainer>
      <Icon src={PersonSvg} alt='user' />
      <SearchInput {...restInputProps} />
      <SearchIcon src={SearchSvg} alt='search' onClick={searchOnClick} />
    </SearchContainer>
  );
};

export default React.memo(Search);