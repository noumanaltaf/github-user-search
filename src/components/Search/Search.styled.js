import styled from "styled-components";

export const SearchContainer = styled.div`
  display: flex;
  align-items: center;
  width: 350px;
  height: 40px;
  border-radius: 40px;
  box-shadow: 0 6px 8px rgb(0 0 0 / 15%);
  background: #fff;
  transition: all 0.3s ease;

  &:focus-within {
    box-shadow: 0 3px 4px rgba(0, 0, 0, 0.15);
  }
`;

export const SearchInput = styled.input`
  outline: none;
  border: none;
  width: calc(100% - 85px);
  height: 30px;
`;

export const Icon = styled.img`
  width: 20px;
  height: 20px;
  margin: 0 10px;
`;

export const SearchIcon = styled(Icon)`
  cursor: pointer;
`