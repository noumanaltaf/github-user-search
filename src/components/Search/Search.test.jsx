import { render } from '@testing-library/react'

import Search from './Search'

describe('Components/Chip', () => {
  it('should render', () => {
    const { container, getByText } = render(<Search placeholder='username' />)

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText('username')).toBeInTheDocument()
  })
})
