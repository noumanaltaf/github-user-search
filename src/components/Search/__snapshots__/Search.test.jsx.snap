// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Chip should render 1`] = `
<div
  class="sc-bczRLJ kBcgRk"
>
  <img
    alt="user"
    class="sc-dkzDqf haWmmY"
    src="person.svg"
  />
  <input
    class="sc-gsnTZi jHfyPU"
    placeholder="username"
  />
  <img
    alt="search"
    class="sc-dkzDqf sc-hKMtZM haWmmY gAKHSM"
    src="search.svg"
  />
</div>
`;
